lib/Client/Util.php

```diff
@@ -26,9 +26,27 @@ public function randInt32() {
         return $this->_rng->int32();
     }
 
-    public function nowMicros() {
+    public static function nowMicros() {
         // Note: microtime returns the current time *in seconds* but with
         // microsecond accuracy (not the current time in microseconds!).
         return floor(microtime(TRUE) * 1000.0 * 1000.0);
     }
+
+    public function pushWithMax(&$arr, $item, $max) {
+        if (!($max > 0)) {
+            $max = 1;
+        }
+
+        $arr[] =  $item;
+
+        // Simplistic random discard
+        $count = count($arr);
+        if ($count > $max) {
```

Comment:
I know you just copied this function over, but it doesn't document the expected usage. It requires that it is used for every insert into the backing array in order for it to work correctly. For example if
```
count($arr) == 32 // true
pushWithMax(&$arr, $item, 30)
count($arr) == 32 // true -- even though we want it to be 30
```

Reply:
good point, my new version fixes this.